src/algebra/continued_fractions/computation/correctness_terminating.lean

```diff
@@ -0,0 +1,218 @@
+/-
+Copyright (c) 2020 Kevin Kappelmann. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Kappelmann
+-/
+import algebra.continued_fractions.computation.translations
+import algebra.continued_fractions.terminated_stable
+import algebra.continued_fractions.continuants_recurrence
+import tactic.lint
+/-!
+# Correctness of Terminating Continued Fraction Computations (`gcf.of`)
+
+## Summary
+
+Let us write `gcf` for `generalized_continued_fraction`. We show the correctness of the
+algorithm computing continued fractions (`gcf.of`) in case of termination in the following sense:
+
+At every step `n : ℕ`, we can obtain the value `v` by adding a specific residual term to the
+convergent `(gcf.of v).convergents n`. The residual term will be zero exactly when the continued
+fraction terminated; otherwise, the residual term will be given by the fractional part stored in
+`gcf.int_fract_pair.stream v n`.
+
+For an example, refer to `gcf.comp_exact_value` and for more information
+about the computation process, refer to `algebra.continued_fraction.computation.basic`.
+
+## Main definitions
+
+- `gcf.comp_exact_value` returns the exact value approximated by the continued fraction `gcf.of v`
+  by adding a residual term as described in the summary.
+
+## Main Theorems
+
+- `gcf.comp_exact_value_correctness_of_stream_eq_some` shows that `gcf.comp_exact_value` indeed
+  returns the value `v` when given the convergent and fractional part as described in the summary.
+- `gcf.of_correctness_of_terminated_at` shows the equality `v = (gcf.of v).convergents n`
+  if `gcf.of v` terminated at position `n`.
+-/
+
+namespace generalized_continued_fraction
+open generalized_continued_fraction as gcf
+
+variables {K : Type*} [discrete_linear_ordered_field K] {v : K} {n : ℕ}
+
+/--
+Given two successive continuants (at positions `n + 1` and `n`) of a continued fraction
+`generalized_continued_fraction.of v` and the fractional part at `int_fract_pair.stream n`,
+this function returns the exact value approximated by the continued fraction, that is `v`.
+
+For example, let `(v : ℚ) := 3.4`. We have:
+  `gcf.int_fract_pair.stream v 0 = some ⟨3, 0.4⟩`, and
+  `gcf.int_fract_pair.stream v 1 = some ⟨2, 0.5⟩`,
+
+Now `(gcf.of v).convergents 1 = 3 + 1/2`, and our fractional term (the residual) at position `2` is
+`0.5`. We hence have `v = 3 + 1/(2 + 0.5) = 3 + 1/2.5 = 3.4`.
+-/
+protected def comp_exact_value (predConts conts : gcf.pair K) (fr : K) : K :=
```

Comment:
```suggestion
protected def comp_exact_value (pred_conts conts : gcf.pair K) (fr : K) : K :=
```

Reply:
Ah, good catch. I also have that naming inconsistency in another few files. I will do a fix for all of them.